spec/GraphQLOverHTTP.md

```diff
@@ -745,3 +745,37 @@ payload is `application/json` then the client MUST NOT rely on the body to be a
 well-formed _GraphQL response_ since the source of the response may not be the
 server but instead some intermediary such as API gateways, proxies, firewalls,
 etc.
+
+# Additional Notes
+
+## Security
+
+In this specification, GET requests are not supported for mutations due to
+security concerns. GET requests expose variables to logging mechanisms and
+intermediaries due to the URL encoding of parameters, which can lead to
+sensitive data being inadvertently logged. Furthermore, GET requests are
+considered "basic requests" under CORS (Cross-Origin Resource Sharing), meaning
+they bypass preflight checks that add a layer of security.
+
+On the other hand, using `application/json` for request bodies mandates a CORS
+preflight request, adding a security layer by ensuring the client has explicit
+permission from the server before sending the actual request. This is
+particularly important in mitigating cross-site request forgery (CSRF) attacks.
+
+Additionally, supporting form data requests (`application/x-www-form-urlencoded`
+or `multipart/form-data`) could pose significant security risks. Form data
+requests may be vulnerable to CSRF and other attacks due to the lack of CORS
+preflight checks. As a result, the use of form data for GraphQL queries or
+mutations is discouraged.
```

Comment:
Not keen on this wording; `multipart/form-data` can be done securely regarding CORS with the right headers enforced by the server. Lots of people successfully make GraphQL queries and mutations with [GraphQL multipart requests](https://github.com/jaydenseric/graphql-multipart-request-spec) in production systems.

The wording "is discouraged" is hostile to the [GraphQL multipart request spec](https://github.com/jaydenseric/graphql-multipart-request-spec), which one day will be moved here into a GraphQL over HTTP spec.

A better approach would be to mention that "simple requests" don't have normal CORS behavior, and to give a few examples of the kinds of requests that are "simple". `multipart/form-data` shouldn't be singled out as somehow worse than any other kind of simple request. In my spec, I don't prescribe what people should do to achieve security, I just note that they should be aware that they need to understand the implications and account for it:

https://github.com/jaydenseric/graphql-multipart-request-spec?tab=readme-ov-file#security

Most people just enforce in their GraphQL server middleware that if the request is `multipart/form-data`, a certain custom header must be present, indicating that the request isn't "simple".

Reply:
+1 Apollo has `Apollo-Require-Preflight` ([doc](https://www.apollographql.com/docs/apollo-server/security/cors/#preventing-cross-site-request-forgery-csrf)). 

I'm not overly familiar with the topic but I would welcome an explicit recommendation in this note to help implementers. `GraphQL-Require-Preflight` maybe?

Reply:
@jaydenseric @martinbonnin I took another pass at this paragraph.  Please review the revised PR - thanks!

Reply:
@jaydenseric Any comments on the revised PR?